docs/userguide/campaigns.md

```diff
@@ -1,25 +1,156 @@
+
+## Getting recommendations
+
+### Basics
+
+```{attention}
+Adding recommendations and measurements using the `recommend` and `add_measurements`
+functions is the safe only way to inform a `Campaign` object about new measurements.
+These functions update the necessary metadata that is crucial for the proper
+execution of a campaign. It is important to rely on these functions to maintain the
+integrity and reliability of the campaign's execution.
+```
+
+To obtain a recommendation for the next experiment, we can query the campaign and use
+the [`recommend`](baybe.campaign.Campaign.recommend) function. The function takes only
+one argument, which is the `batch_quantity` keyword. This specifies the desired size of
+the batch of experiments to be conducted.
+
+```python
+rec = campaign.recommend(batch_quantity=3)
+```
+
+The `recommend` function returns a `DataFrame` with `batch_quantity` many rows, each 
+representing a set of parameters from the search space.
+
+### Influence of the batch size
+
+Regarding the choice of a suitable `batch_size`, it is important to understand the
+difference between performing multiple recommendations with batch size of 1 and a single
+recommendation with a larger batch size.
+* **Larger batch size**: When using a larger batch size, the recommended experiments are
+chosen to *jointly* optimized the acquisition function.
+This means that the recommendations are made considering the interaction of multiple
+experiments together.
+* **Smaller batch size**: When making several smaller recommendations, each *individual*
+recommendation optimizes the acquisition function at the specific point in time when it
+is requested. In this case, the recommendations are made independently of each other
+without considering the joint optimization.
+
+### Caching of recommendations
+
+Whenever recommendations are made, the `Campaign` object caches them. If measurements
+for the recommendations are added, then the cached recommendations are deleted. However,
+if no measurements are added and the `recommend` function is called again, then the
+`Campaign` object simply returns the cached recommendations instead of generating new
+ones. This caching mechanism helps to optimize performance by avoiding unnecessary
+re-computations when measurements are not provided.
+
+## Adding measurements
+
+To add measurements, we expand the  `DataFrame` that was created by the `recommend`
+function by adding a new column for the target. We can then provide the campaign with
+these measurements and receive a new recommendation.
+
+```python
+rec["quality"] = [2,4,9]
+campaign.add_measurements(rec)
+new_rec = campaign.recommend(batch_quantity=5)
+```
+
+In discrete or hybrid search space, numerical measurements are required to fall into a
+predefined tolerance by default.
+This tolerance is defined on the level of the individual parameters.
+This requirement can be disabled by using the `numerical_measurements_must_be_within_tolerance` flag.
```

Comment:
its not so easy to see where this flag should go, this is another example why I prefere a code block incluyding all relevant optionals like
```python
bla = Campaign(
   searchspace=space,
   objective=obj,
   strategy=strat,  # optional, see ...
   numerical_measurements_must_be_within_tolerance = False,  # optional
)
```

Reply:
@Scienfitz I do not agree with your proposed solution of having a single call that shows all of the optional arguments: This is what the actual doc is for, and in my opinion, the user guide should guide the users how to use a Campaign with a MWE.
Regardless of that point, I have changed the formulation. If you are happy with the change of formulation here (independent of the issue whether we do want a minimalistic or full construction of a campaign here) please resolve. For that issue, I guess it makes more sense to open a separate conversation here.

Reply:
what is the rational you dont prefer this? These two additional pseudo-code lines would easily make sure 
1. the user is aware of optionals and 
2. is not only aware of their existence but also **where** they'd belong and even 
3. user can get an idea of what the expected input format is (trivial in this case but for instance in the targets user guide I can tell the user that for some enums both the singleton or the string is accepted). 
4. does all of that in a super compact way
These are all big advantages imo. 

I only see disadvantages for putting these in a text below, so I am curious for your reasoning.

Reply:
My main worry is that we overload the user with information when we directly tell them about everything immediately. But as it seems that you have a very strong opinion on this, I will change this once it is clear how to deal with pseudo-code, which will then in fact become necessary if we want to have it in a minimal way.

Reply:
sure we shouldnt overwhelm, but do you **really** see that danger here?
here a further example of the usefulness of the proposed:
![image](https://github.com/emdgroup/baybe/assets/17951239/6b638223-5f8c-4a04-8b9a-8af289b42164)

How is the user overwhelmed by two lines of pseudo code (with comment) versus several new lines of text below and potentially having to fly to other pages? Seems the latter is much more overwhelming to me (and could potentially be overread)